plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/views/FilterDialog.java

```diff
@@ -0,0 +1,714 @@
+						contrastApplications.add(app);
+						count++;
+						ContrastUIActivator.logInfo(application.getName());
+					}
+				}
+			} else {
+				List<Application> apps = server.getApplications();
+				for (Application application : apps) {
+					ApplicationUIAdapter app = new ApplicationUIAdapter(application, application.getName());
+					contrastApplications.add(app);
+					count++;
+					ContrastUIActivator.logInfo(application.getName());
+				}
+			}
+		}
+		ApplicationUIAdapter allApplications = new ApplicationUIAdapter(null, "All Applications(" + count + ")");
+		contrastApplications.add(allApplications);
+		applicationCombo.setInput(contrastApplications);
+
+		if (setSavedDefaults) {
+			IEclipsePreferences prefs = ContrastCoreActivator.getPreferences();
+			String appId = prefs.get(Constants.APPLICATION_ID, Constants.ALL_APPLICATIONS);
+			ApplicationUIAdapter selected = allApplications;
+			for (ApplicationUIAdapter adapter : contrastApplications) {
+				if (appId.equals(adapter.getId())) {
+					selected = adapter;
+					break;
+				}
+			}
+			applicationCombo.setSelection(new StructuredSelection(selected));
+		}
+	}
+
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		Composite container = (Composite) super.createDialogArea(parent);
+
+		GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+
+		Composite comboComposite = new Composite(container, SWT.NONE);
+		comboComposite.setLayout(new GridLayout(2, false));
+
+		label = new Label(comboComposite, SWT.NONE);
+		gd = new GridData(SWT.FILL, SWT.CENTER, false, false);
+		label.setLayoutData(gd);
+		label.setText("Server");
+		String orgUuid = getOrgUuid();
+
+		createServerCombo(comboComposite, orgUuid);
+		updateServerCombo(orgUuid, true, servers);
+
+		serverCombo.addSelectionChangedListener(serverComboBoxListener);
+
+		label = new Label(comboComposite, SWT.NONE);
+		gd = new GridData(SWT.FILL, SWT.CENTER, false, false);
+		label.setLayoutData(gd);
+		label.setText("Application");
+
+		createApplicationCombo(comboComposite, orgUuid);
+		updateApplicationCombo(orgUuid, true, applications);
+
+		createSeverityLevelSection(container);
+
+		createLastDetectedSection(container);
+
+		createStatusSection(container);
+
+		populateFiltersWithDataFromEclipsePreferences();
+
+		return container;
+	}
+
+	private void createSeverityLevelSection(Composite container) {
+
+		GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+
+		Composite severityCompositeContainer = new Composite(container, SWT.NONE);
+		severityCompositeContainer.setLayout(new GridLayout(2, false));
+
+		label = new Label(severityCompositeContainer, SWT.NONE);
+		gd = new GridData(SWT.FILL, SWT.CENTER, false, false);
+		label.setLayoutData(gd);
+		label.setText("Severity");
+
+		Composite severityComposite = new Composite(severityCompositeContainer, SWT.NONE);
+		severityComposite.setLayout(new GridLayout(3, false));
+
+		severityLevelNoteButton = new Button(severityComposite, SWT.CHECK);
```

Comment:
Create a function where you can init the Button object and pass the required variables as parameters to recycle code. If you use these kind of procedure in other classes you should put it on a Util class.

Reply:
Done.